src/components/common/Button/index.tsx

```diff
@@ -114,21 +115,18 @@ type CustomProps = {
   isLoading?: boolean;
   to?: string | undefined;
   variant?: ResponsiveValue<keyof typeof variants>;
-  sizes?: ResponsiveValue<keyof typeof sizes>;
+  buttonSize?: ResponsiveValue<keyof typeof sizes>;
 } & SystemProps;
 
 type Props<
   T extends ElementType = typeof _defaultElement
 > = PolymorphicPropsWithRef<CustomProps, T>;
 
-const StyledButton = styled(Box)
-  .withConfig({
-    shouldForwardProp: (prop) => !['sizes'].includes(prop),
-  })
-  .attrs((props) => ({
```

Comment:
forwarding button still gives trouble with the as prop. It doesn't forward the styling props when using `as`. It does forward the props when using `forwardedAs` but the typing support is based on `as`.